crawl-ref/source/mon-death.cc

```diff
@@ -2295,8 +2312,13 @@ item_def* monster_die(monster& mons, killer_type killer,
             corpse = daddy_corpse;
     }
     if (mons.has_ench(ENCH_BOUND_SOUL))
+    {
         _make_derived_undead(&mons, !death_message, true);
 
+    if (!mons.is_summoned() && mons.type == MONS_PHARAOH_ANT)
+        _pharaoh_ant_bind_souls(&mons);
+    }
```

Comment:
These `{` `}` look like a bug. Do you really want `_pharaoh_ant_bind_souls` to only trigger if the pharaoh ant itself has ENCH_BOUND_SOUL?